sdk/core/core-rest-pipeline/src/util/userAgentPlatform-browser.mts

```diff
@@ -66,7 +66,7 @@ function getBrandVersionString(brands: BrowserBrand[]): BrowserBrand | undefined
 export async function setPlatformSpecificData(map: Map<string, string>): Promise<void> {
   const localNavigator = globalThis.navigator as NavigatorEx;
   let osPlatform = "unknown";
-  if (localNavigator.userAgentData) {
+  if (localNavigator?.userAgentData) {
```

Comment:
Out of curiosity, in what environment did you encounter this issue (Node, browser, something else...)? Seems strange that the browser mapping is getting loaded in an environment where `navigator` is undefined.

Cc @mpodwysocki

Reply:
yes, all browsers should support `navigator`. Unless the code is running on NodeJS or some other environment but browser version got pulled in/bundled?

Reply:
@BrandonStudio Right for NodeJS we would expect this file being used instead https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/core-rest-pipeline/src/util/userAgentPlatform.ts#L25

do you have a simplified project or steps that reproduce the issue you are seeing?

Reply:
@jeremymeng are your missing the Edge Runtime ?

Reply:
@arvinxx Edge Runtime is not on our official supported platform, but thanks for bring it up.  We will try our best to resolve blocking issues for customers.

Reply:
@BrandonStudio our core-rest-pipeline package [uses the conditional exports](https://nodejs.org/docs/latest-v20.x/api/packages.html#community-conditions-definitions)

https://github.com/Azure/azure-sdk-for-js/blob/7b02c7275fb1d20c9d95f77ad017b51b042ba28d/sdk/core/core-rest-pipeline/package.json#L10-L34

I'd expect NodeJS to use `import` or `require`, not the `browser` one.  But there are probably some platforms that use mixture of NodeJs and browsers APIs, for example, Electron can enable NodeJS API by setting `nodeIntegration` to true. 

Also it's possible that some tools doesn't work properly with conditional exports.

We also have fallback fields for legacy platforms via `main` and `browser` entries
https://github.com/Azure/azure-sdk-for-js/blob/7b02c7275fb1d20c9d95f77ad017b51b042ba28d/sdk/core/core-rest-pipeline/package.json#L7-L8

Reply:
Is any bundler being used? Depending on how the bundler is configured, it could be that the bundler is pulling in the browser conditional export. If that's happening, and the bundled code is then being run on Node, that might cause this behavior.

Reply:
Looks that Vercel allows users to choose between "nodejs" or "edge" runtime.  In edge runtime, there are only [some Web compatible apis](https://edge-runtime.vercel.app/features/available-apis) available.  I couldn't see any more telemetry information beyond a `globalThis.EdgeRuntime` which has the value of "edge-runtime".  Given that we also use `localNavigator?.platform` down below, I think it is safe to merge this PR for now. We can add edge runtime platform in a separate PR. @timovv what do you think?

Reply:
Sounds good, no objection to merging the PR here, good to have this as a sanity check.

Maybe this Edge runtime pulls in the browser platform entry by default?

Reply:
> Maybe this Edge runtime pulls in the browser platform entry by default?

yes in my debugging I see browser version was used.